### DISCUSSION TITLE
how to get all my products and update their prices?

### DISCUSSION THREAD

**Discussion by mscampos92** [saleweaver/python-amazon-sp-api#39] [Q&A]
First of all, nice work! :)

My doubt is about the best way to get all my products(catalog?) to update their prices if I need...

Thank you! :)

**Comment by saleweaver** [comment]
There are many ways to do this, and it depends largely on the use case. Do you want to create an automated repricer to update the prices once a competitor changes theirs, or what exactly is it you want to build?

To get all the products, use the reports api and get (for example) an active listings report (See reportTypes enum for all possible values).  

```
Reports(marketplace=Marketplaces.US).create_report(
        reportType=ReportType.GET_MERCHANT_LISTINGS_ALL_DATA)
```

You can get current prices using the `Products` client then, simply pass it skus. If you want to react to pricing changes I'd use the notifications API instead though, Amazon sends a message for every price change to one of your competitor's prices. 

Once you know what price you want to set, you can create a xml file, I believe it's the Pricing Feed, and use the `Feeds` client to upload the file 

```
Feeds(marketplace...).submit_feed(feed_type, file, content_type='application/xml')

**Reply by mscampos92** [reply]
Basicly I have to update some products when I receive any "new prices lists" from any provider.. so.. it's not very "automated job" .. 

ok.. I got the report... but it downloaded a file.. how it works? :)

**Reply by saleweaver** [reply]
Then you don't really need the products as long as your skus are in the supplier's files. You can simply create the xml and upload a feed.

**Reply by mscampos92** [reply]
yes... but in other marketplaces I do some "checks" before send some update.. for example.. I really need to update this price? or the change is just 10 cents... that's why I want to get all my products from amazon... any idea?

**Reply by saleweaver** [reply]
The listings or prices report will do this.

**Reply by mscampos92** [reply]
thanks for your help.
now I got it.

I need to use 3 "steps"

create_report(passing the report type)
get_report(passing reportId)
get_report_document(passing reportDocumentId)

**Reply by saleweaver** [reply]
Yes exactly. You can schedule report too
